requests/utils.py

```
@@ -635,11 +635,14 @@ def get_auth_from_url(url):
     """Given a url with authentication components, extract them into a tuple of
     username,password."""
     if url:
-        url = unquote(url)
         parsed = urlparse(url)
-        return (parsed.username, parsed.password)
-    else:
-        return ('', '')
+
+        try:
+            return (unquote(parsed.username), unquote(parsed.password))
+        except (AttributeError, TypeError):
+            pass
```

I know I'm guilty of them, but I'm not the biggest fan of `except: pass` statements. Perhaps this could be rewritten as:

``` python
try:
    auth = (unquote(parsed.username), unquote(parsed.password))
except (AttributeError, TypeError):
    auth = ('', '')

return auth
```

But, you know, with the proper indentation. ;)